### DISCUSSION TITLE
Official MWC in maintenance. Some findings with regards to beercss from over there.

### DISCUSSION THREAD

**Discussion by gobijan** [beercss/beercss#342] [General]
Hi,

as you might have heard Google's Material Components for the web were set to maintenance mode due to resource constraints.
I participated over there and endorsed BeerCSS as alternative.

User @KTibow pointed out some deviations from the material standard as he has implemented Material 3 for Svelte (https://ktibow.github.io/m3-svelte/).

You can find the discussion here: https://github.com/material-components/material-web/discussions/5642#discussioncomment-10171475

His main findings are those:

**-Dialog**

Can't close with ESC, and doesn't lock focus to within itself

**-Extended FAB**
Non-standard behavior of showing text only on hover

**-Progress**
In both bar and circular, the end isn't rounded

**-Search**
Bugged, you can't type anything as you lose focus as soon as you try

**-Text input**
Missing text cursor in edges (admittedly official material libs did this previously so)

**-Slider**
Missing click/drag cursor
Incorrect rounding of handle
Handle stays "active" until you click somewhere else
Probably more stuff I didn't check, such as spacing anomalies

**-General: Elements like navigation rail, drawer, list, icon button, etc**
Non-standard behavior of handling hover by showing items as selected instead of hovered

**-General: fonts**
Roboto isn't loaded
Letter spacing doesn't change by font location

**-General: interactivity**
Elevation effect is too simple
No transitions in many places
Between pressing your mouse and releasing your mouse, the ripple disappears
The ripple (including hover/focus states) is a fixed color
The ripple always comes from the center
Some things are clickable but aren't buttons (bad for accessibility)
No focus outline (bad for accessibility)

**-General: shapes**
Endorses a lot of non-standard stuff, like odd rounding of elements, changing the background or outline of a component, using more blur and shadows, etc

What you guys think about those items? Valid?

**Comment by leonardorafael** [comment]
Hi @gobijan ,

There is a valid issue in this list, about search. I think this could be impacted by the latest release. The others are just hating. This is not the first time this user write hater comments about us, with misinformation and trying to minimize our work (and maximize his work). I could discuss all points here and there, but this is a lost of time. I just breathe up, take a beer and go ahead 😮‍💨🍺😅

There could be some changes in official Material Design 3, that we could not up to date. This could be done soon. Just need some time to work on it.

I will open an issue about search 👍

Thanks for the post!

**Reply by gobijan** [reply]
cheers 🍺

**Reply by KTibow** [reply]
Thanks to both of you for bringing this up. I apologize if my comments came across as hating or attempting to minimize Beer CSS - that was never my intention. My goal was solely to suggest places where it could be improved to be more consistent with Material 3.

Regarding these comments, I'd be happy to explain and discuss any of my points that you find inaccurate. I do appreciate the work you're doing with Beer CSS, and I'm looking forward to a more consistent version.

**Reply by leonardorafael** [reply]
Hi guys,

I think I owe some explanations here. The main purpose of this CSS framework is to build UIs, in the simplest way possible, using just HTML + CSS. If some of the behaviors can be done with HTML + CSS then we will do it. For this reason, many of the ones mentioned above are inaccurated for this project, committed only to HTML + CSS. And yet for this reason, there are some small differences in the implementation. In fact, Google's own systems have small differences in the implementation of fonts, effects, elevations... and they are still Material Design 3.

Why do we prefer to define more basic behaviors?

Generally speaking, when we increase the complexity of the behaviors (none of them may perform the desired behavior) and all that added behavior ends up making it difficult to meet the needs that the app actually needs. For example, to close a dialog we can press ESC, Backspace, browser/device back button, drag our fingers down/side, use a voice command. After opening a dialog, we can block the scroll, focus on the dialog, focus on the button or focus on the first field (blank or filled?) and also do nothing (for waiting dialogs). There is a wide variety of behaviors, so we prefer to focus on the simple and what can be done with just CSS and HTML. This explains why there are some inaccurated points, perhaps because it is not clear where we want to go. 

So back to the points:

Inaccurate points (may be accurate point if it can be done with HTML + CSS only):

- Dialogs
- Extended FABs (we have the `active` helper that shows the text in the event you want)
- Progress
- General: fonts (by default, we use the system's fonts, on Android will use Roboto for example, without the need to download any fonts)
- General: interactivity
- General: shapes (yes we want to go further, we don't want to limit or restrict ourselves)

Accurated points (some of them might just be out of date):
- Search (really in trouble here)
- Slider (this is an exception, we had to use JS to change the value and I'm not happy with it yet)
- Text input
- General: Elements like navigation rail, drawer, list, icon button, etc.

The JS file of Beer CSS is almost optional. Basically is a JS helper for the dynamic theme and the trigger of elements. More about here https://github.com/beercss/beercss/blob/main/docs/JAVASCRIPT.md. Despite choosing for simplicity, it is not a CSS framework that only delivers simple, restricted and pre-made things. You can scale your UI just combining the things. It's fun. We have some fun here https://codepen.io/leo-bnu

Hope this can clarify something 👍

**Reply by KTibow** [reply]
I see. However, I do want to let you know that you could consider other options:
- Dialogs: Try using the [dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) tag. Other frameworks use it, and it can work without JS if you use `<form>`s.
- Progress: Recently, the appearance of progress elements changed. AFAIK nobody has implemented them fully to spec on the web yet, so that could be a first.
![image](https://github.com/user-attachments/assets/6dd691d1-bc52-4ea3-a408-3b879f252a05)

I'm also going to *attempt* to make some PRs that change some things. If they seem good, undraft and approve them. If not just close them.
